v1/bytes.go

```diff
@@ -0,0 +1,43 @@
+package v1
+
+var zeroBytes = ByteValue{value: 0, unit: Byte}
+
+type ByteValue struct {
+	// Value is the whole non-negative number of bytes of the specified unit
+	value uint32
+
+	// Unit is the unit of the byte value
+	unit ByteUnit
+}
+
+func NewByteValue(value int32, unit ByteUnit) ByteValue {
+	if value < 0 {
+		return zeroBytes
+	}
+	return ByteValue{
+		value: uint32(value),
+		unit:  unit,
+	}
+}
```

Comment:
thoughts on
```suggestion
var zeroBytes = byteValue{value: 0, unit: Byte}

type ByteValue interface {
	Value() uint32
	Unit() ByteUnit
}

type byteValue struct {
	// Value is the whole non-negative number of bytes of the specified unit
	value uint32

	// Unit is the unit of the byte value
	unit ByteUnit
}

func NewByteValue(value int32, unit ByteUnit) ByteValue {
	if value < 0 {
		return zeroBytes
	}
	return byteValue{
		value: uint32(value),
		unit:  unit,
	}
}

func (b byteValue) Value() uint32 {
	return b.value
}

func (b byteValue) Unit() ByteUnit {
	return b.unit
}

```
so that we don't have to expose the struct and callers _must_ use `NewByteValue`

Reply:
I like it